src/ViewModel/CheckoutConfigProvider.php

```diff
@@ -68,12 +78,16 @@ public function getConfig(): string
         $storeCode = $checkoutConfig['storeCode'];
         $checkoutConfig['payment']['restUrlPrefix'] = "/rest/$storeCode/V1/";
 
-        return $this->serializer->serialize([
-            'storeCode' => $storeCode,
-            'payment' => $checkoutConfig['payment'],
-            'language' => $this->localeResolver->getLocale(),
-            'currency' => $this->currencyProvider->getConfig(),
-            'defaultCountryId' => $checkoutConfig['defaultCountryId'],
+        $transport = new \Magento\Framework\DataObject([
+          'storeCode' => $storeCode,
+          'payment' => $checkoutConfig['payment'],
+          'language' => $this->localeResolver->getLocale(),
+          'currency' => $this->currencyProvider->getConfig(),
+          'defaultCountryId' => $checkoutConfig['defaultCountryId'],
         ]);
+
+        $this->eventManager->dispatch('hyva_react_checkout_config', ['transport' => $transport]);
```

Comment:
It is good we pass `$checkoutConfig` as well. Because there are occasion where we need to pass other data from the `$checkoutConfig` as well. So that will hep to avoid a dependency injection in the observer in order to load it?

Reply:
Agree! I've just updated the PR to include it as well.